src/main/java/dataFactory/GameDataMapper.java

```diff
@@ -0,0 +1,28 @@
+package dataFactory;
+
+import data.GameData;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Row;
+
+import java.util.function.Function;
+
+/**
+ * GameDataMapper is a function that maps an Excel Row to a GameData object.
+ * It extracts the title, age, and price from the first three cells of the row.
+ * The row index is also captured for reporting purposes.
+ */
+public class GameDataMapper implements Function<Row, GameData> {
+    @Override
+    public GameData apply(Row row) {
+        Cell titleCell = row.getCell(0);
```

Comment:
Hmm. Reading cell data by index make your code is hard to maintenance in case of column change in the future